OpenSim/Common/Test/CMakeLists.txt

```diff
@@ -15,5 +15,5 @@ OpenSimAddTests(
     TESTPROGRAMS ${TEST_PROGS}
     DATAFILES ${TEST_FILES} ${C3D_TEST_FILES} ${TRC_TEST_FILES} 
               ${MOT_TEST_FILES}
-    LINKLIBS ${BTK_LIBRARIES} osimCommon ${SIMTK_ALL_LIBS} 
+    LINKLIBS ${BTK_LIBRARIES} osimCommon osimSimulation ${SIMTK_ALL_LIBS} 
```

Comment:
I suggest opening a new issue to augment testComponentInterface with Reporter-related tests. There are currently no tests for ConsoleReporter or TableReporter, and this PR adds tests in practical use cases, which I think is valuable.